k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageViewFragment.java

```diff
@@ -741,26 +744,34 @@ public void onMessageDataLoadFinished(LocalMessage message) {
 
             displayHeaderForLoadingMessage(message);
             mMessageView.setToLoadingState();
+            showProgressThreshold = 0L;
         }
 
         @Override
         public void onMessageDataLoadFailed() {
             Toast.makeText(getActivity(), R.string.status_loading_error, Toast.LENGTH_LONG).show();
+            showProgressThreshold = 0L;
         }
 
         @Override
         public void onMessageViewInfoLoadFinished(MessageViewInfo messageViewInfo) {
             showMessage(messageViewInfo);
+            showProgressThreshold = 0L;
         }
 
         @Override
         public void onMessageViewInfoLoadFailed(MessageViewInfo messageViewInfo) {
             showMessage(messageViewInfo);
+            showProgressThreshold = 0L;
         }
 
         @Override
         public void setLoadingProgress(int current, int max) {
-            mMessageView.setLoadingProgress(current, max);
+            if (showProgressThreshold == 0L) {
+                showProgressThreshold = System.nanoTime() + PROGRESS_THRESHOLD_NANOS;
```

Comment:
[`SystemClock.elapsedRealtime()`](https://developer.android.com/reference/android/os/SystemClock.html#elapsedRealtime()) is probably the better choice on Android.